statistics/histogram.go

```diff
@@ -1312,6 +1312,20 @@ func (c *Column) GetColumnRowCount(sctx sessionctx.Context, ranges []*ranger.Ran
 	return rowCount, nil
 }
 
+// ItemID implements TableCacheItem
+func (c *Column) ItemID() int64 {
+	return c.Info.ID
+}
+
+// DropEvicted implements TableCacheItem
+// DropEvicted drops evicted structures
+func (c *Column) DropEvicted() {
+	if c.StatsVer < Version2 {
+		c.CMSketch = nil
+		c.Loaded = false
```

Comment:
The comment says "Loaded means if the histogram, the topn and the cm sketch are loaded fully". Before the pr, we assume either all three are loaded or none of the three are loaded. Now we break the assumption. It may bring some problems. For example, `(*Column).IsLoaded` returns `c.Loaded || c.notNullCount() > 0`. It seems that we assume `c.Loaded` is always equal to `c.notNullCount() > 0` and will remove `c.notNullCount() > 0` in the future. Now if the cm sketch is evicted in ver1 case, `c.Loaded` is false while `c.notNullCount() > 0` is true. cc @time-and-fate

Reply:
the loaded usage in column now is revised

Reply:
@xuyifangreeneyes is correct.
There're mainly two usages of `IsLoaded`:
1. help to check if we can use the non-pseudo estimation logic for this `Column`.
2. help to check if stats in this `Column` needs loading.

And of course, they should be the same before this PR.
It seems that they become different in this PR, so we may need to introduce a new field or change `IsLoaded` to represent more states to handle this.

Reply:
the revised loaded updated in https://github.com/pingcap/tidb/pull/35361